src/ngx_http_lua_api.c

```diff
@@ -194,13 +198,12 @@ ngx_http_lua_shared_memory_init(ngx_shm_zone_t *shm_zone, void *data)
 
     lmcf->shm_zones_inited++;
 
-    if (lmcf->shm_zones_inited == lmcf->shm_zones->nelts
-        && lmcf->init_handler)
+    if (lmcf->shm_zones_inited == lmcf->shm_zones->nelts)
     {
         saved_cycle = ngx_cycle;
         ngx_cycle = ctx->cycle;
 
-        rc = lmcf->init_handler(ctx->log, lmcf, lmcf->lua);
+        rc = ngx_http_lua_delay_init_handler();
```

Comment:
@agentzh YES. If there is no `lua_shared_dict`, or we can say there is no one declare delay `init_by_lua`(I've moved all delay `init_by_lua to use ngx_http_lua_delay_init_handler to implement, include original `lua_shared_dict`). Then, the `lmcf->delay_init` is `null`, `init_by_lua` will run in ngx_lua's postconfiguration, that is [here](https://github.com/openresty/lua-nginx-module/pull/1050/files#diff-c91b2ec425e9be9d56fee2df662b67d2R746).